tests/test_bsfs.c

```diff
@@ -1181,6 +1181,165 @@ START_TEST(test_read_large_buf) {
 }
 END_TEST
 
+const char* truncate_str = "saljbhajfhaojefbgaof";
+bs_file_t truncate_file;
+
+static void truncate_fs_setup(void) {
+  int fd = create_tmp_file(FS_DISK_SIZE + 0x4000);
+  ck_assert_int_eq(bsfs_init(fd, &tmp_fs), 0);
+
+  stego_key_t truncate_key;
+  ck_assert_int_eq(stego_gen_user_keys(&truncate_key, 1), 0);
+  ck_assert_int_eq(keytab_store(tmp_fs->disk, 0, "truncatelvl", &truncate_key),
+                   0);
+
+  void* zero = calloc(1, BFT_SIZE);
+  ck_assert(zero);
+  ck_assert_int_eq(bft_write_table(&truncate_key, tmp_fs->disk, zero), 0);
+  ck_assert_int_eq(fs_write_bitmap(&truncate_key, tmp_fs->disk, zero), 0);
+  free(zero);
+
+  ck_assert_int_eq(bsfs_mknod(tmp_fs, "truncatelvl/file", S_IFREG), 0);
+
+  ck_assert_int_eq(bsfs_open(tmp_fs, "truncatelvl/file", &truncate_file), 0);
+  ck_assert_int_eq(
+      bsfs_write(truncate_file, truncate_str, strlen(truncate_str), 0),
+      strlen(truncate_str));
+}
+
+static void truncate_fs_teardown(void) {
+  bsfs_destroy(tmp_fs);
+}
+
+START_TEST(test_ftruncate_extend) {
+  ck_assert_int_eq(bsfs_ftruncate(truncate_file, 30), 0);
+
+  uint8_t buf[10];
+  uint8_t expected[sizeof(buf)] = { 0 };
+  ck_assert_int_eq(bsfs_read(truncate_file, buf, 10, 20), 10);
```

Comment:
When using the size of `truncate_file`, it should be  `strlen(truncate_str)` and not a number like 20 or 30.

Reply:
Resolved in 2ad81ae.